### ISSUE TITLE
DataStore random failures

### ISSUE THREAD

**Issue by PleaseStopAsking** [Esri/arcgis-powershell-dsc#197]
Found that the ``ArcGIS_DataStore`` resource will randomly return a failure but checking ArcGIS Server and DataStore shows a successful connection. 

From the below output, the issue seems to start on line 551 which is making a call to the ``Get-DataStoreTypesToRegister`` function which is then calling the ``Get-DataStoreInfo`` function which is then calling ``Invoke-ArcGISWebRequest``.

https://github.com/Esri/arcgis-powershell-dsc/blob/037b14af2d058f54c96a0f46f4facd9baf77579f/Modules/ArcGIS/DSCResources/ArcGIS_DataStore/ArcGIS_DataStore.psm1#L551

I have been unable to reproduce it every time and it seems to be random at best so I have been unable to trace it down any further at this time.

**Set Output:**

```
VERBOSE: [2019-07-29 17:41:11Z] [VERBOSE] [foo]: LCM:  [ Start  Set      ]  
[[ArcGIS_DataStore]arcgisDatastore]
VERBOSE: [2019-07-29 17:41:11Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Importing the module ArcGIS_DataStore in force mode.
VERBOSE: [2019-07-29 17:41:11Z] [WARNING] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] The names of some imported commands from the module 'ArcGISUtility' include 
unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the 
Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
VERBOSE: [2019-07-29 17:41:12Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Current value of property hostidentifier is '' and hostidentifierType is ''
VERBOSE: [2019-07-29 17:41:12Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Configured host identifier '' does not match expected value 
'foo.bar.local' or host identifier type '' does not match expected value 'hostname'. Setting it
VERBOSE: [2019-07-29 17:41:12Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Hostidentifier.properties file was modified. Need to restart the 'ArcGIS Data 
Store' service to pick up changes
VERBOSE: [2019-07-29 17:41:12Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Current value of property failover_on_primary_stop is false
VERBOSE: [2019-07-29 17:41:12Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Property 'failover_on_primary_stop' will be modified. Need to restart the 'ArcGIS 
Data Store' service to pick up changes
VERBOSE: [2019-07-29 17:41:12Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Stop Service 'ArcGIS Data Store' before applying property change
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Stopping the service
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Stopped the service
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Adding entry failover_on_primary_stop = true to 
C:\ArcGIS\DataStore\framework\etc\datastore.properties
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Updating file C:\ArcGIS\DataStore\framework\etc\datastore.properties
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Changed applied:- True
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Restarting Service 'ArcGIS Data Store' to pick up property change
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Restarted Service 'ArcGIS Data Store'
VERBOSE: [2019-07-29 17:41:13Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Waiting for Url 
https://foo.bar.local:2443/arcgis/datastoreadmin/configure?f=json
VERBOSE: [2019-07-29 17:41:15Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [Warning]:- Exception calling "GetResponse" with "0" argument(s): "Unable to 
connect to the remote server"
VERBOSE: [2019-07-29 17:41:36Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [Warning]:- Exception calling "GetResponse" with "0" argument(s): "The operation 
has timed out"
VERBOSE: [2019-07-29 17:41:42Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Url is OK
VERBOSE: [2019-07-29 17:41:43Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Waiting for Url https://foo.bar.local:6443/arcgis/admin
VERBOSE: [2019-07-29 17:41:43Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Url is OK
VERBOSE: [2019-07-29 17:41:43Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Ensure the Publishing GP Service (Tool) is started on Server
VERBOSE: [2019-07-29 17:41:43Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Service Status :- @{configuredState=STARTED; realTimeState=STARTED}
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Checking if Relational Datastore is registered
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] The machine foo with FQDN 'foo.bar.local' does NOT 
participates in a registered 'Relational' data store
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Registering datastores Relational
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Adding Relational as a data store type
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore at https://localhost:2443/arcgis/datastoreadmin with DataStore 
Content directory at H:\arcgisdatastore for server https://foo.bar.local:6443/arcgis
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore Attempt 1
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore on Machine foo.bar.local
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] POST https://localhost:2443/arcgis/datastoreadmin/configure with 236-byte payload
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] received -1-byte response of content type application/json;charset=UTF-8
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Response:- {"error":{"code":500,"message":"Unable to configure the ArcGIS Data 
Store with the GIS Server. Please make sure that the GIS Server URL is accessible, the account specified has 
administrative privileges to the site, and the publishing tools is started on the GIS Server.","details":null}}
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Error Response - @{code=500; message=Unable to configure the ArcGIS Data Store with
 the GIS Server. Please make sure that the GIS Server URL is accessible, the account specified has administrative 
privileges to the site, and the publishing tools is started on the GIS Server.; details=}
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [WARNING]:- ERROR: failed. Unable to configure the ArcGIS Data Store with the GIS 
Server. Please make sure that the GIS Server URL is accessible, the account specified has administrative privileges to 
the site, and the publishing tools is started on the GIS Server.
VERBOSE: [2019-07-29 17:41:45Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Attempt [1] Failed. Retrying after 30 seconds
VERBOSE: [2019-07-29 17:42:15Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore Attempt 2
VERBOSE: [2019-07-29 17:42:15Z] Settings handler status to 'transitioning' 
(C:\Packages\Plugins\Microsoft.Powershell.DSC\2.77.0.0\Status\3.status)
VERBOSE: [2019-07-29 17:42:15Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Checking if datastore is registered
VERBOSE: [2019-07-29 17:43:55Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [WARNING]:- Invalid URI: The hostname could not be parsed.
VERBOSE: [2019-07-29 17:43:55Z] Settings handler status to 'transitioning' 
(C:\Packages\Plugins\Microsoft.Powershell.DSC\2.77.0.0\Status\3.status)
VERBOSE: [2019-07-29 17:43:55Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Attempt [2] Failed. Retrying after 30 seconds
VERBOSE: [2019-07-29 17:44:25Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore Attempt 3
VERBOSE: [2019-07-29 17:44:25Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Checking if datastore is registered
VERBOSE: [2019-07-29 17:46:05Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [WARNING]:- Invalid URI: The hostname could not be parsed.
VERBOSE: [2019-07-29 17:46:05Z] Settings handler status to 'transitioning' 
(C:\Packages\Plugins\Microsoft.Powershell.DSC\2.77.0.0\Status\3.status)
VERBOSE: [2019-07-29 17:46:05Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Attempt [3] Failed. Retrying after 30 seconds
VERBOSE: [2019-07-29 17:46:35Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore Attempt 4
VERBOSE: [2019-07-29 17:46:35Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Checking if datastore is registered
VERBOSE: [2019-07-29 17:48:16Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [WARNING]:- Invalid URI: The hostname could not be parsed.
VERBOSE: [2019-07-29 17:48:16Z] Settings handler status to 'transitioning' 
(C:\Packages\Plugins\Microsoft.Powershell.DSC\2.77.0.0\Status\3.status)
VERBOSE: [2019-07-29 17:48:16Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Attempt [4] Failed. Retrying after 30 seconds
VERBOSE: [2019-07-29 17:48:46Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Register DataStore Attempt 5
VERBOSE: [2019-07-29 17:48:46Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] Checking if datastore is registered
VERBOSE: [2019-07-29 17:50:26Z] [VERBOSE] [foo]:                            
[[ArcGIS_DataStore]arcgisDatastore] [WARNING]:- Invalid URI: The hostname could not be parsed.
VERBOSE: [2019-07-29 17:50:26Z] Settings handler status to 'transitioning' 
(C:\Packages\Plugins\Microsoft.Powershell.DSC\2.77.0.0\Status\3.status)
VERBOSE: [2019-07-29 17:50:26Z] [VERBOSE] [foo]: LCM:  [ End    Set      ]  
[[ArcGIS_DataStore]arcgisDatastore]  in 554.7740 seconds.
```

**Comment by shailesh91** [comment]
@PleaseStopAsking I have identified the issues. Parameter ServerURL is using the wrong variable "ServerURL" instead of "ServerSiteURL" in the following code snippet.
https://github.com/Esri/arcgis-powershell-dsc/blob/037b14af2d058f54c96a0f46f4facd9baf77579f/Modules/ArcGIS/DSCResources/ArcGIS_DataStore/ArcGIS_DataStore.psm1#L551-L553

**Comment by shailesh91** [comment]
Fixed with v2.1.1!
